### PR TITLE
Signups endpoint 2

### PR DESCRIPTION
**Pull request by chloealee**

### CODE REVIEW

File: lib/modules/dosomething/dosomething_api/includes/Transformer.php [hunk 1]
```diff
@@ -412,15 +412,25 @@ protected function transformReportbackItem($data) {
...
+      'campaign_run' => [
+        'id' => $data->campaign_run,
+        'current' => $current,
+        'timing' => [
+          'start' => 'start-date-placeholder',
+          'end' => 'end-date-placeholder',
+        ],
+      ],
       'uri' => $data->uri,
+      'reportbacks' => $data->reportbacks,
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_api/includes/Transformer.php]: Reportbacks should be included in the `SignupTransformer` after calling `$data['campaign'] = $this->transformCampaign((object) $item->campaign);` and sending to the `$this->transformReportback()` method.

File: lib/modules/dosomething/dosomething_signup/includes/SignupTransformer.php [hunk 2]
```diff
@@ -61,16 +61,27 @@ public function show($id) {
    * @return array
    */
   protected function transform($item) {
+    // print_r($item);
+    // die();
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_signup/includes/SignupTransformer.php]: debug!

File: lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc [hunk 3]
```diff
@@ -143,3 +143,26 @@ function dosomething_reportback_filter_flagged_reportbacks($params) {
...
+ *
+ * @param string $run_id Campaign run id. 
+ * @param string $sid Signup id. 
+ * @return mixed
+ */
+function dosomething_reportback_filter_reportbacks_by_signups($run_id, $sid) {
+  $reportbacks = db_query("SELECT DISTINCT s.sid, r.rbid, r.run_nid, s.run_nid
+                          FROM dosomething_signup s 
+                          INNER JOIN dosomething_reportback r on s.run_nid = r.run_nid
+                          WHERE s.uid = r.uid;");
```
Comment (weerd) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc]: So this feels like its on the right track, but you could likely do it all in the `dosomething_signup.query.inc` file, so it's all in one hit to the DB. Just add it to the query there with a join on the conditions :) Then in the **SignupTransformer.php** you can do  `$this->reportbacks = Reportback::get($rbid)` without the try catch cause it should return just one :)

Comment (chloealee) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc]: awesome, thanks Diego! Makes so much more sense! One question - could a person technically have multiple reportbacks associated with one signup?

Comment (chloealee) [inline lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc]: Also, `dosomething_reportback_filter_reportbacks_by_signups` does some logic to get rid of the reportbacks with `reportback_run` of `0` and return a user's reportbacks by signup id. 

I'm not sure where to put that logic - it seems like it'd be too much going on if we placed after `$result = $query->execute();` on line 28 in `dosomething_signup.query.inc`?

**Comment by chloealee** [comment]
Closing this PR - continuing work in #6168.

**Pull request closed without merging**